### PR TITLE
scheduler: consider evict leader when calc expect

### LINKED ISSUE #3959: hot-region-scheduler not work when there is evict-leader-scheduler in cluster
### What did you do?

test hot-region-scheduler and evict-leader-scheduler

### What did you expect to see?

work

### What did you see instead?

hot-region-scheduler not work
image

### What version of PD are you using (`pd-server -V`)?

master

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

fix #3959

### What is changed and how it works?

1. consider evict leader when calc expect
2. reduce the pending amp factor
3. unify qps and query as query, to ensure consistency with the documentation https://github.com/pingcap/docs-cn/pull/6835

### Check List

Tests

- Unit test

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn): https://github.com/pingcap/docs-cn/pull/6835
- Need to cherry-pick to the release branch
-  - 4.0 5.0 5.1 5.2

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/pdctl/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -270,7 +270,7 @@ func (s *schedulerTestSuite) TestScheduler(c *C) {
 		"minor-dec-ratio":            0.99,
 		"src-tolerance-ratio":        1.05,
 		"dst-tolerance-ratio":        1.05,
-		"read-priorities":            []interface{}{"qps", "byte"},
+		"read-priorities":            []interface{}{"query", "byte"},
```
Comment (Yisaer) [inline tests/pdctl/scheduler/scheduler_test.go:273]: The `QueryPriority` is `qps` while there is `query` here.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

File: server/schedulers/utils.go [hunk 2]
```diff
@@ -640,15 +640,16 @@ func summaryStoresLoadByEngine(
...
-	// Stores without byte rate statistics is not available to schedule.
 	for _, info := range storeInfos {
 		store := info.Store
 		id := store.GetID()
 		storeLoads, ok := storesLoads[id]
 		if !ok || !collector.Filter(info) {
 			continue
 		}
-
+		if !info.Store.AllowLeaderTransfer() && kind == core.LeaderKind {
```
Comment (HunDunDM) [inline server/schedulers/utils.go:650]: How about adding this in `collector.Filter`?

**Pull request merged**